### PR TITLE
Add processor aliases back to -print-supported-cpus and -mcpu=help

### PR DESCRIPTION
**Pull request by jroelofs**
They were accidentally dropped in https://github.com/llvm/llvm-project/pull/96249

rdar://140853882

### CODE REVIEW

File: clang/test/Driver/print-supported-cpus.c [hunk 1]
```diff
@@ -25,3 +26,12 @@
...
 // CHECK-ARM: Use -mcpu or -mtune to specify the target's processor.
+
+// RUN: %clang --target=arm64-apple-macosx --print-supported-cpus 2>&1 | \
+// RUN:   FileCheck %s --check-prefix=CHECK-AARCH64 --implicit-check-not=apple-latest
+
+// CHECK-AARCH64: Target: arm64-apple-macosx
+// CHECK-AARCH64: apple-m1
+// CHECK-AARCH64: apple-m2
+// CHECK-AARCH64: apple-m3
+// CHECK-AARCH64: Use -mcpu or -mtune to specify the target's processor.
```
Comment (jroelofs) [inline clang/test/Driver/print-supported-cpus.c:37]: - [ ] FIXME: EOF newline

**Review by jthackray — APPROVED** [review]
LGTM

**Review by tommat01 — APPROVED** [review]
I'm surprised how complicated this was to do, but LGTM.

File: clang/test/Driver/print-supported-cpus.c (hunk 1 above)
Comment (fhahn) [inline clang/test/Driver/print-supported-cpus.c:37]: nit: might be worth to move this to a separate file, to not further restrict the configs where this test can be run? might also be good to check for a CPU that't not aliased, e.g. apple-m4?

File: llvm/lib/MC/MCSubtargetInfo.cpp [hunk 2]
```diff
@@ -104,14 +110,17 @@ static void Help(ArrayRef<SubtargetSubTypeKV> CPUTable,
...
   unsigned MaxFeatLen = getLongestEntryLength(FeatTable);
 
   // Print the CPU table.
   errs() << "Available CPUs for this target:\n\n";
-  for (auto &CPU : CPUTable)
-    errs() << format("  %-*s - Select the %s processor.\n", MaxCPULen, CPU.Key,
-                     CPU.Key);
+  for (auto &CPUName : CPUNames) {
+    if (CPUName == "apple-latest")
+      continue;
```
Comment (fhahn) [inline llvm/lib/MC/MCSubtargetInfo.cpp:120]: Could you add a comment here why this is skipped?

File: llvm/lib/MC/MCSubtargetInfo.cpp [hunk 3]
```diff
@@ -137,8 +146,11 @@ static void cpuHelp(ArrayRef<SubtargetSubTypeKV> CPUTable) {
 
   // Print the CPU table.
   errs() << "Available CPUs for this target:\n\n";
-  for (auto &CPU : CPUTable)
-    errs() << "\t" << CPU.Key << "\n";
+  for (auto &CPU : CPUNames) {
+    if (CPU == "apple-latest")
+      continue;
+    errs() << "\t" << CPU << "\n";
+  }
```
Comment (fhahn) [inline llvm/lib/MC/MCSubtargetInfo.cpp:153]: unrelated to this patch directly, but might be good to share the logic with `Help` in the future?

**Review by fhahn — APPROVED** [review]
LGTM, thanks.

Add a few minor suggestions

**Pull request merged**